### PR TITLE
feature/list-structure

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS PROJECT I:

1. Used webpack to bundle JavaScript.
2. Learnt how to use proper ES6 syntax.
3. Used ES6 modules to write modular JavaScript.
4. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
5. Made sure that I used correct [GitHub Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
6. Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const todos = todoTasks.map((todo) => `
+  <div class="todoTasks">
+    <div class="task"><input type="checkbox" id="checkBox" name="checkbox" value="1">
+      <label for="checkBox">${todo.type}</label>
+    </div>
+    <i class="fa fa-ellipsis-v" aria-hidden="true"></i>
+  </div>
+`).join('');
+
```
Comment (V-Blaze) [inline src/index.js]: - [ ] You did great so far, but I think there are some issues you need to fix before moving to the next milestone, 
- [ ] Looking at the video, you can see that your application starts up with a webpack compilation issue with reference to the `mode`, even after closing the error, you can it gives an error in the console and the todo items are not shown in the UI, I'll suggest you rectify this issue. :+1: :+1: :100: 

- - VIDEO REFERENCE:

https://github.com/Kgomotso196/To-Do-List/assets/107632780/3df5ddf0-84d0-4c6e-a6df-676953df4ab2

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Kgomotso196 , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Linters setup correctly✔
- Readme looks professional👋 

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import './style.css';
+
+const todoTasks = [
+  { index: 1, type: 'Code Development', completed: true },
+  { index: 2, type: 'Code Testing', completed: false },
+  { index: 3, type: 'Code Maintenance', completed: true },
+];
+
```
Comment (julie-ify) [inline src/index.js:8]: - [ ] The tasks are not being displayed in the UI, as pointed out by the previous reviewer. You can consider using the browser for clues on errors.

  image

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Kgomotso196,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ishimwezachee — APPROVED** [review]
Hi @Kgomotso196 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image
## Highlights
Good feature branch name ✔️
No linter errors ✔️
Descriptive commit messages ✔️
Descriptive README ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**